### ISSUE TITLE
`no-navigation-withour-resolve`: Track function return values

### ISSUE THREAD

**Issue by marekdedic** [sveltejs/eslint-plugin-svelte#1486]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

latest

### What version of `eslint-plugin-svelte` are you using?

master

### What did you do?

```svelte
<scrip>
  const resolveBlogUrl = (slug: string) => resolve('/blog/[slug]', { slug });
</script>

<a href={resolveBlogUrl('my-post')}>Link to My Post</a>
```

### What did you expect to happen?

No error

### What actually happened?

Error

### Link to **GitHub Repo** with Minimal Reproducible Example

Will make a PR

**Comment by marekdedic** [comment]
To be honest, I'm a little bit on the fence with this one. @sacrosanctic @ota-meshi thoughs? If the user had TS enabled, than they are covered by #1321, so this would be just for non-TS users...

**Comment by ota-meshi** [comment]
I think it would be useful to be able to perform checks that do not rely on type information when that information is not provided.
